mycore-ocfl/src/main/java/org/mycore/ocfl/MCROCFLXMLClassificationManager.java

```diff
@@ -0,0 +1,332 @@
+        lastModified = new Date(MCRCategoryDAOFactory.getInstance().getLastModified(mcrid.getRootID()));
+
+        try (InputStream objectAsStream = xml.getInputStream()) {
+            VersionInfo versionInfo = buildVersionInfo(message, lastModified);
+            getRepository().stageChanges(ObjectVersionId.head(objName), versionInfo, updater -> {
+                updater.writeFile(objectAsStream, buildFilePath(mcrCg), OcflOption.OVERWRITE);
+            });
+        } catch (IOException e) {
+            throw new MCRPersistenceException("Failed to update object '" + objName + "'", e);
+        } catch (IllegalArgumentException e) {
+            // Issue seems to be with the OCFL Library trying to find a file it deleted itself
+            // LOGGER.warn("IllegalArgumentException happened, Ignoring...");
+
+            // dropChanges(mcrid);
+            // LOGGER.error("Something has gone Wrong!", e);
+        }
+
+    }
+
+    public void fileDelete(MCRCategoryID mcrid, MCRCategory mcrCg, MCRContent clXml, MCRContent cgXml,
+        MCREvent eventData) {
+        String objName = getName(mcrid);
+        String message = eventData.getEventType();
+        Date lastModified = new Date();
+        lastModified = new Date(MCRCategoryDAOFactory.getInstance().getLastModified(mcrid.getRootID()));
+        VersionInfo versionInfo = buildVersionInfo(message, lastModified);
+        getRepository().stageChanges(ObjectVersionId.head(objName), versionInfo, updater -> {
+            updater.removeFile(buildFilePath(mcrCg));
+        });
+    }
+
+    public void fileMove(Map<String, Object> data, MCREvent eventData) {
+        MCRCategoryImpl oldParent = (MCRCategoryImpl) data.get("ctg");
+        MCRCategoryImpl newParent = (MCRCategoryImpl) eventData.get("parent");
+        int index = (int) eventData.get("index");
+        MCRCategoryImpl mcrCg = (MCRCategoryImpl) data.get("ctg");
+        fileDelete(mcrCg.getId(), mcrCg, (MCRContent) data.get("xml"), eventData);
+        mcrCg.setParent(newParent);
+        List<MCRCategory> children = newParent.getChildren();
+        children.remove(mcrCg);
+        children.add(index, mcrCg);
+        MCRContent newParentXml = new MCRJDOMContent(MCRCategoryTransformer.getMetaDataElement(newParent, USE_COUNTER));
+        MCRContent oldParentXml = new MCRJDOMContent(MCRCategoryTransformer.getMetaDataElement(oldParent, USE_COUNTER));
+        fileUpdate(newParent.getId(), newParent, newParentXml, eventData);
+        fileUpdate(oldParent.getId(), oldParent, oldParentXml, eventData);
+    }
+
+    public void commitChanges(MCREvent evt) {
+        Map<String, Object> data = MCROCFLEventHandler.getEventData(evt);
+        MCRCategoryID mcrid = (MCRCategoryID) data.get("mid");
+        MCRCategory mcrCg = (MCRCategory) data.get("ctg");
+        MCRContent rtXml = (MCRContent) data.get("rtx"); // class
+        MCREvent event = (MCREvent) evt.get("event");
```

Comment:
Do not use magic Strings here. Make them "public static final" somewhere and document them. What is that "event" within an event anyway?